java/core/src/main/java/com/whylogs/core/DataTypes.java

```diff
@@ -0,0 +1,68 @@
+package com.whylogs.core;
+
+import java.lang.reflect.Type;
+import java.util.HashSet;
+
+public enum DataTypes {
+  Numerical {
```

Comment:
The DataTypes Enum works as a type mapper given we are using the type system, but this could be changed a bit to have classes that function similarly if we believe users will need more customization.